flocker/node/testtools.py

```diff
@@ -4,14 +4,43 @@
 Testing utilities for ``flocker.node``.
 """
 
-from unittest import skipIf
-from subprocess import Popen
+import errno
+import os
+import socket
+from unittest import skipUnless
 
+from ._docker import BASE_DOCKER_API_URL
 from ..testtools import loop_until
 
-# This is terible (https://clusterhq.atlassian.net/browse/FLOC-85):
-if_docker_configured = skipIf(Popen([b"docker", b"version"]).wait(),
-                              "Docker must be installed and running.")
+SOCKET_PATH = BASE_DOCKER_API_URL.split(':/')[-1]
+
+
+def docker_accessible():
+    """
+    Attempt to connect to the Docker control socket.
+
+    This may address https://clusterhq.atlassian.net/browse/FLOC-85.
+
+    :return: ``True`` if the current user has permission to connect, else
+        ``False``.
+    """
+    try:
+        socket.socket(family=socket.AF_UNIX).connect(SOCKET_PATH)
+    except socket.error as e:
+        if e.errno == errno.EACCES:
+            return False
+        if e.errno == errno.ENOENT:
+            # Docker is not installed
+            return False
+        raise
+    else:
+        return True
+
+if_docker_configured = skipUnless(
```

Comment:
This now duplicates https://github.com/ClusterHQ/flocker/commit/6c0dcb8c980c35f7cfb4989b1dbd82fd14a6c778, so we should get rid of the other one (and just use this).